strictdoc/export/excel/excel_generator.py

```diff
@@ -1,59 +1,193 @@
+
+        return refs
+
+    def _init_columns_width() -> int:
+        """Init the width of each column based on the definition + return the defaullt width for the parent column
+
+        Returns:
+            int: Width of the Parent column
+        """
+        for column in EXPORT_COLUMNS:
+            column[MAX_WIDTH_KEY] = (
+                len(column[COLUMN_HEADER_KEY]) + HEADER_MARGIN
+            )
+        parent_header_len = len(PARENT_COLUMN_HEADER_LABEL) + HEADER_MARGIN
+
+        return parent_header_len
+
+    @staticmethod
+    def _set_columns_width(
+        workbook: Workbook, worksheet: Worksheet, parent_header_len: int
+    ):
+        """Set the width of each columns based on the max lenght and the max possible
+
+        Args:
+            workbook (Workbook): Excel workbook
+            worksheet (Worksheet): Excel worksheet
+            parent_header_len (int): Len of the parent column
+        """
+        cell_format_text_wrap = workbook.add_format()
+        cell_format_text_wrap.set_text_wrap()
+
+        for idx, column in enumerate(EXPORT_COLUMNS, start=0):
+            if column[MAX_WIDTH_KEY] > MAX_WIDTH:
+                worksheet.set_column(idx, idx, MAX_WIDTH, cell_format_text_wrap)
+            else:
+                worksheet.set_column(idx, idx, column[MAX_WIDTH_KEY])
+        worksheet.set_column(
+            len(EXPORT_COLUMNS), len(EXPORT_COLUMNS), parent_header_len
+        )
+
+    @staticmethod
+    def _init_headers() -> List:
+        """Init the header list
+
+        Returns:
+            List: List of header
+        """
+        headers: List = []
+
+        for column in EXPORT_COLUMNS:
+            headers.append({"header": column[COLUMN_HEADER_KEY]})
+        headers.append({"header": PARENT_COLUMN_HEADER_LABEL})
+
+        return headers
+
     @staticmethod
     def export_tree(
         document_tree: DocumentTree, traceability_index, output_excel_root
     ):
+        """Will create and fill one excel file per document, containing requierements with an UID
+
+        Args:
+            document_tree (DocumentTree): List of document to export
+            traceability_index (Any): traceability)index
+            output_excel_root (str): Path of the export
+        """
         Path(output_excel_root).mkdir(parents=True, exist_ok=True)
 
         document: Document
         for document in document_tree.document_list:
-            document_out_file = "{}/{}.xlsx".format(
-                output_excel_root, document.meta.document_filename_base
+            document_out_file = Path(
+                output_excel_root,
+                f"{document.meta.document_filename_base}.xlsx",
             )
 
-            workbook = xlsxwriter.Workbook(document_out_file)
-            worksheet = workbook.add_worksheet()
+            document_contents = list(
+                DocumentCachingIterator(document).all_content()
+            )  # not ideal but needed to make the first pass to detect refs
```

Comment:
This work should have already been done at this point. Please check the APIs of `traceability_index.get_document_iterator()`. It equally applies to my own commit that you are building on top of.